lib/internal/encoding.js

```diff
@@ -319,6 +323,23 @@ class TextEncoder {
     return encodeUtf8String(`${input}`);
   }
 
+  encodeInto(src, dest) {
+    validateEncoder(this);
+    validateString(src, 'src');
+
+    const iterator = src[Symbol.iterator]();
+    let item = iterator.next();
+    const items = [];
+    while (!item.done) {
+      items.push(item.value);
+      item = iterator.next();
+    }
```

Comment:
Is this equivalent to `const items = [...src];`?

And am I correct in assuming that the reason for this is to avoid splitting surrogate pairs? Because I can see that in that case this is an easy approach, but it also seems like it might not be great in terms of performance?